source/SIL.AppBuilder.Portal.Frontend/src/ui/components/inputs/locale-select/index.tsx

```diff
@@ -22,39 +22,54 @@ export type IProps = IOwnProps & WithDataProps & i18nProps;
 const translationMap = { 'en-US': 'en', 'es-419': 'es', 'fr-FR': 'fr' };
 
 class LocaleSelect extends React.Component<IProps> {
+  defaultValue = '';
+
+  componentDidMount() {
+    this.defaultToFirst();
+  }
+
+  componentDidUpdate() {
+    this.defaultToFirst();
+  }
+
+  defaultToFirst = () => {
+    const { onChange, selected } = this.props;
+    if (!selected) {
+      onChange(this.defaultValue);
+    }
+  };
+
   onSelect = async (e, { value }) => {
     e.preventDefault();
 
-    const { i18n } = this.props;
-    const { currentUser, dataStore } = this.props;
+    const { onChange, selected } = this.props;
 
-    i18n.changeLanguage(value);
+    if (value === selected) {
+      return;
+    }
 
-    await update(dataStore, currentUser, {
-      attributes: { locale: value },
-    });
+    onChange(value);
   };
 
   render() {
-    const { currentUser, i18n } = this.props;
+    const { currentUser, i18n, onChange } = this.props;
     const attributes = attributesFor(currentUser) as UserAttributes;
     const userLocale = attributes.locale;
     const { options, language } = i18n;
     const languages = Object.keys(options.resources);
 
-    const selected = userLocale || language;
-
+    this.defaultValue = userLocale || language;
```

Comment:
in react components, we aren't supposed to use class properties to manage state (React is the only JS library that I know of to require this)

you'll need to use state / setState.

Some info here: https://reactjs.org/docs/state-and-lifecycle.html#using-state-correctly

Reply:
Redid this significantly, including the state change requests mentioned here